components/apimgt-extensions/org.wso2.carbon.apimgt.webapp.publisher/src/test/java/org/wso2/carbon/apimgt/webapp/publisher/APIPublisherServiceTest.java

```diff
@@ -0,0 +1,100 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.apimgt.webapp.publisher;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+import org.wso2.carbon.apimgt.integration.client.OAuthRequestInterceptor;
+import org.wso2.carbon.apimgt.integration.client.model.OAuthApplication;
+import org.wso2.carbon.apimgt.webapp.publisher.dto.ApiScope;
+import org.wso2.carbon.apimgt.webapp.publisher.dto.ApiUriTemplate;
+import org.wso2.carbon.apimgt.webapp.publisher.exception.APIManagerPublisherException;
+
+import java.lang.reflect.Field;
+import java.util.*;
+
+/**
+ * This is the test class for {@link APIPublisherServiceImpl}
+ */
+public class
+APIPublisherServiceTest extends BaseAPIPublisherTest {
+    private static final Log log = LogFactory.getLog(APIPublisherServiceTest.class);
+    private APIPublisherServiceImpl apiPublisherService = new APIPublisherServiceImpl();
+    private APIConfig apiConfig = new APIConfig();
+
+    @BeforeTest
+    public void initialConfigs() throws Exception {
+        setApiConfigs(apiConfig);
+        initialise0OAuthApplication();
```

Comment:
is the method name "initialise0OAuthApplication", "0" is intended?? I don't think that is needed, we can simply say "initializeOAuthApplication"